code/datums/materials/alloys.dm

```diff
@@ -0,0 +1,163 @@
+/** Materials made from other materials.
+  */
+/datum/material/alloy
+	name = "alloy"
+	desc = "A material composed of two or more other materials."
+	material_flags = NONE
+	/// The materials this alloy is made from weighted by their ratios.
+	var/list/composition = null
+	/// Breakdown flags required to reduce this alloy to its component materials.
+	var/req_breakdown_flags = BREAKDOWN_ALLOYS
+
+/datum/material/alloy/return_composition(amount=1, breakdown_flags)
+	if(req_breakdown_flags & !(breakdown_flags & req_breakdown_flags))
+		return ..()
+
+	. = list()
+	var/list/cached_comp = composition
+	for(var/comp_mat in cached_comp)
+		var/datum/material/component_material = SSmaterials.GetMaterialRef(comp_mat)
+		var/list/component_composition = component_material.return_composition(cached_comp[comp_mat], breakdown_flags)
+		for(var/comp_comp_mat in component_composition)
+			.[comp_comp_mat] += component_composition[comp_comp_mat] * amount
+
+
+/** Plasteel
+  *
+  * An alloy of iron and plasma.
+  * Applies a significant slowdown effect to any and all items that contain it.
+  */
+/datum/material/alloy/plasteel
+	name = "plasteel"
+	desc = "The heavy duty result of infusing iron with plasma."
+	color = "#706374"
+	material_flags = MATERIAL_INIT_MAPLOAD
+	value_per_unit = 0.135
+	strength_modifier = 1.25
+	integrity_modifier = 1.5 // Heavy duty.
+	armor_modifiers = list(MELEE = 1.4, BULLET = 1.4, LASER = 1.1, ENERGY = 1.1, BOMB = 1.5, BIO = 1, RAD = 1.5, FIRE = 1.1, ACID = 1)
+	sheet_type = /obj/item/stack/sheet/plasteel
+	categories = list(MAT_CATEGORY_RIGID=TRUE, MAT_CATEGORY_BASE_RECIPES=TRUE)
+	composition = list(/datum/material/iron=1, /datum/material/plasma=1)
+
+/datum/material/alloy/plasteel/on_applied_obj(obj/item/target_item, amount, material_flags)
+	. = ..()
+	if(!isitem(target_item))
+		return
+
+	target_item.slowdown += MATERIAL_SLOWDOWN_PLASTEEL * amount / MINERAL_MATERIAL_AMOUNT
+
+/datum/material/alloy/plasteel/on_removed_obj(obj/item/target_item, material_flags)
+	. = ..()
+
+	if(!isitem(target_item))
+		return
+
+	target_item.slowdown = initial(target_item.slowdown)
```

Comment:
This can cause edge-cases when multiple sources affect slowdown.

Reply:
I'm not sure how to fix this without implementing a slowdown modifier system for items. Do you have any advice because that seems a bit out of scope?

Reply:
Could you instead just remove the slowdown again? ofcourse also not ideal but I think it has less edgecases.

Reply:
I've removed the slowdown modifier. It's a significant buff which is the *opposite* of what we're trying to do right now, I'll try to think of something else.

Reply:
I think you misunderstood, I meant that you do target_item.slowdown -= (whatever calcs you did to add it)

Reply:
instead of setting it to initial

Reply:
I'll do that.

Reply:
Should be done, I had to add proper amount handling to all of the material removal procs. it looks like bananium and possibly hot ice already had it. \<sarcasm>I'm sure that wouldn't have broken all if anything ever bothered to remove materials.\</sarcasm>